### PR TITLE
test(backup/s3): additional integration tests

### LINKED ISSUE #10184: S3 backup store tests
In addition to our existing tests for the S3 backup store we should add some more:

**S3 Compatibility** 
- [ ] S3 backup store is compatible with MinIO (can be automated via testcontainers)
- [ ] S3 backup store is compatible with AWS S3 (manually)

**Failure Modes**
- [ ] Corrupting a backup prevents restore

**Zeebe Compatibility**
- [ ] Backups without snapshot are supported
- [ ] Backups with real snapshots and segments are supported

### PR DESCRIPTION
**Pull request by lenaschoenburg**
* Adjusts the backup descriptor to allow for backups without snapshot
* All tests run for backups with and without snapshots
* All tests are run on Localstack and MinIO
* Organizes tests into multiple nested classes
* Adds another test case to show that deleting objects prevents a restore

closes #10184

### CODE REVIEW

**Comment by oleschoenburg** [comment]
@deepthidevaki It's a large PR but most of it is just restructuring of existing tests

File: backup/src/main/java/io/camunda/zeebe/backup/api/BackupDescriptor.java [hunk 1]
```diff
@@ -15,7 +17,7 @@ public interface BackupDescriptor {
   /**
    * @return id of the snapshot included in the backup
    */
-  String snapshotId();
+  Optional<String> snapshotId();
```
Comment (deepthidevaki) [inline backup/src/main/java/io/camunda/zeebe/backup/api/BackupDescriptor.java:20]: Thank you.

**Review by deepthidevaki — APPROVED** [review]
Great! :rocket:

**Pull request merged**